File: src/main/java/edu/cit/estillore/MentorMatch/MentorMatch/subject/Subject.java
package edu.cit.estillore.MentorMatch.MentorMatch.subject;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * A subject/skill that tutors can be matched against (e.g. "Calculus I").
 * Managed by admins (FR-012); referenced by TutorSubject.
 */
@Entity
@Table(name = "subjects", uniqueConstraints = {
        @UniqueConstraint(columnNames = "name")
})
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Subject {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String name;

    @Column(length = 255)
    private String description;
}

File: android/app/src/main/java/edu/cit/estillore/mentormatch/data/model/Subject.kt
package edu.cit.estillore.mentormatch.data.model

import com.squareup.moshi.JsonClass

/** Mirrors SubjectController.java's Subject entity. */
@JsonClass(generateAdapter = true)
data class Subject(
    val id: Long,
    val name: String,
    val description: String? = null
)
